### DISCUSSION TITLE
Is there a way to read many files?

### DISCUSSION THREAD

**Discussion by gkim132** [jspsych/jsPsych#962] [Q&A] [answered]
Hi,

I need to use a lot of image files stored in several folders. 
I'm wondering if there is a way to read all files easily instead of listing all the file names. 
what I mean listing all the file names is: imgSet_1 = ["a/1.png","a/2.png","a/3.png", ... ]
imgSet_2 = ["b/1.png","b/2.png","b/3.png", ... ]
 ...
imgSet_10 = [...]

Any help will be greatly appreciated!

Thank you!

**Comment by vijaymarupudi** [comment, marked as answer]
Due to security reasons, browsers cannot do this. However if you know how to use a programming language, this can be easier.

In R, after you install or verify that you have the `jsonlite` package, you can do this and copy paste the output.

```R
jsonlite::toJSON(list.files("/path/to/folder", recursive=TRUE))
```

**Comment by ineminor** [comment]
Hello! I am also encountering this problem now, did you solve it?

**Reply by ChristopheBossens** [reply]
Hi ineminor,

As vijaymarupudi mentioned, this is not possible using Javascript and jsPsych itself. In stead, you use another programming language such as R or Python to list all the files in a folder such as your image folder. You then create a json string which you copy paste into your javascript code.

To make this more specific, the following array in your Javascript code: 

` ["a/1.png","a/2.png","a/3.png", ... ]`

is a string that was generated in another programming language.
